src/controllers/competition.controller.ts

```diff
@@ -137,22 +148,37 @@ export class CompetitionController {
       // Get all teams (excluding admin teams)
       const teams = await services.teamManager.getAllTeams(false);
       
-      // Map team IDs to names
-      const teamMap = new Map(teams.map(team => [team.id, team.name]));
+      // Filter out disqualified teams and track if any were removed
+      const activeTeams = teams.filter(team => !team.disqualified);
+      const teamMap = new Map(activeTeams.map(team => [team.id, team]));
       
-      // Format leaderboard with team names
-      const formattedLeaderboard = leaderboard.map((entry, index) => ({
-        rank: index + 1,
-        teamId: entry.teamId,
-        teamName: teamMap.get(entry.teamId) || 'Unknown Team',
-        portfolioValue: entry.value
-      }));
+      // Track disqualified teams and create filtered leaderboard
+      const disqualifiedTeamIds = new Set(
+        teams
+          .filter(team => team.disqualified)
+          .map(team => team.id)
+      );
+      
+      const filteredLeaderboard = leaderboard.filter(entry => !disqualifiedTeamIds.has(entry.teamId));
+      const hasFilteredTeams = filteredLeaderboard.length < leaderboard.length;
+      
+      // Format leaderboard with team names, recalculating ranks
+      const formattedLeaderboard = filteredLeaderboard.map((entry, index) => {
```

Comment:
Seems to me that here, you could simply created the formatted and filtered leaderboard mapping in one go, and then simply check the map.size and compare with the original to get hasFilteredTeams, no?

Reply:
agreed - added: [982e4fe](https://github.com/recallnet/trading-simulator/pull/10/commits/982e4fea91d2b98c3914423b2339bbf067318235)